js/content-slider.js

```diff
@@ -1,85 +1,81 @@
-      controlsRight.style.height    = `${contentBlockUpperRect.height}px`;
+      this.contentSliderControls.style.top = `${this.controlSurfaceRect.top - intitialTopValue}px`;
 
-      // console.log("initial contentBlockUpperRect.top",contentBlockUpperRect.top);
-      // console.log("initial contentSliderControlsRect.top", contentSliderControlsRect.top);
-
-      // viewport starts at [0, 79px] ???
-      let intitialTopValue = contentSliderControlsRect.top; //79px
-      let diff = 0;
-
-      contentSliderControls.style.top = `${contentBlockUpperRect.top - intitialTopValue}px`;
-      
-      let resizeObserver = new ResizeObserver(entries => {
-        for (let entry of entries){
-        //   recalculate the controls rect.
-          contentSliderControlsRect = contentSliderControls.getBoundingClientRect();
+    } else {
+      console.error(
+        "[Content Slider] No content slider found. Please make sure the .className parameter passed to the content slider exists and is correct."
+      );
+    }
+  }
 
-          //find the diff between controlRect and upper rect 
-          diff = contentBlockUpperRect.top - contentSliderControlsRect.top;
+  resizeControlSurface() {
+    this.controlSurfaceRect = this.controlSurface.getBoundingClientRect();
+    this.contentSliderControlsRect = this.contentSliderControls.getBoundingClientRect();
 
-        //  Adjust controls' top - math is incorect
-          // contentSliderControls.style.top = `${contentBlockUpperRect.top - intitialTopValue + diff}px`;
-        }
-      });
-      
-      resizeObserver.observe(this.contentSlider);
+    this.controlsLeft.style.height = `${this.controlSurfaceRect.height}px`;
+    this.controlsRight.style.height = `${this.controlSurfaceRect.height}px`;
+  }
 
-      //event listeners
-      controlsLeft.addEventListener("click", (e) => {
-        e.preventDefault();
-        contentSliderCardsList.scrollLeft -= 500;
-        
-      })
+  initializeControlSurface() {
+    this.contentSliderCardsList = this.contentSlider.querySelector(
+      ".content-slider__cards-list"
+    );
+    this.contentSliderControls = this.contentSlider.querySelector(
+      ".content-slider__controls"
+    );
+    this.controlsLeft = this.contentSliderControls.querySelector(
+      ".content-slider__controls-left"
+    );
+    this.controlsRight = this.contentSliderControls.querySelector(
+      ".content-slider__controls-right"
+    );
+  }
 
-      controlsRight.addEventListener("click", (e) => {
-        e.preventDefault();
-        contentSliderCardsList.scrollLeft += 500;
+  attachEventListeners() {
+    // TODO: Refactor to a single function with behavior based on event.someValue
+    // TODO: Verify desired scroll on click behavior with design; 525px ensures at least one card is in view
+    this.controlsLeft.addEventListener("click", (e) => {
+      e.preventDefault();
+      this.contentSliderCardsList.scrollLeft -= 525; //px scroll amount
```

Comment:
The big improvement I could see here is to never assume the 525 number. I'd look at just getting the width of the `.content-block` as it's styled, and use that as the shift amount, plus or minus margin or whatever. Then you could have sliders with larger or smaller elements and it would just work.